### PR TITLE
Pods 3432 PSA Subscription- Company UK - replace page to improve collection of company number.

### PR DESCRIPTION
**Pull request by grantrobson**

### CODE REVIEW

File: app/controllers/register/company/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -48,11 +48,14 @@ class CheckYourAnswersController @Inject()(
   def onPageLoad(mode: Mode): Action[AnyContent] = (authenticate andThen allowAccess(mode) andThen getData andThen requireData) {
     implicit request =>
 
+      implicit val ua: UserAnswers = request.userAnswers
```
Comment (suryabalasingh) [inline app/controllers/register/company/CheckYourAnswersController.scala]: no need to pass implicit user answers

File: app/identifiers/register/company/CompanyRegistrationNumberId.scala [hunk 2]
```diff
@@ -17,7 +17,26 @@
...
+import play.api.i18n.Messages
+import utils.UserAnswers
+import utils.checkyouranswers.{BooleanCYA, CheckYourAnswers, CheckYourAnswersCompany, StringCYA}
+import viewmodels.{AnswerRow, Link}
 
 case object CompanyRegistrationNumberId extends TypedIdentifier[String] {
+  self =>
   override def toString: String = "companyRegistrationNumber"
+
+  implicit def cya(implicit userAnswers: UserAnswers, messages: Messages): CheckYourAnswers[self.type] =
```
Comment (suryabalasingh) [inline app/identifiers/register/company/CompanyRegistrationNumberId.scala]: implcit useranswers can be removed as it is already there in row

File: app/utils/navigators/RegisterCompanyNavigator.scala [hunk 3]
```diff
@@ -193,4 +200,23 @@ class RegisterCompanyNavigator @Inject()(
...
+    match {
+      case Some(BusinessType.UnlimitedCompany) =>
+        NavigateTo.dontSave(routes.HasCompanyCRNController.onPageLoad(NormalMode))
+      case Some(BusinessType.LimitedCompany) =>
+        NavigateTo.dontSave(routes.CompanyRegistrationNumberController.onPageLoad(NormalMode))
+      case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
+    }
+  }
+  private def hasCompanyCRNNavigation(answers:UserAnswers):Option[NavigateTo] = {
+    answers.get(HasCompanyCRNId)
```
Comment (suryabalasingh) [inline app/utils/navigators/RegisterCompanyNavigator.scala]: this logic is duplicate

Comment (grantrobson) [reply app/utils/navigators/RegisterCompanyNavigator.scala]: done

File: app/identifiers/register/company/CompanyRegistrationNumberId.scala (hunk 2 above)
Comment (grantrobson) [reply app/identifiers/register/company/CompanyRegistrationNumberId.scala]: done

File: app/controllers/register/company/CheckYourAnswersController.scala (hunk 1 above)
Comment (grantrobson) [reply app/controllers/register/company/CheckYourAnswersController.scala]: done

File: test/controllers/register/company/CheckYourAnswersControllerSpec.scala [hunk 4]
```diff
@@ -276,15 +301,24 @@ object CheckYourAnswersControllerSpec extends ControllerSpecBase {
   }
 
   private def testRenderedView(sections: Seq[AnswerSection], dataRetrievalAction: DataRetrievalAction): Unit = {
+
+
+    def writeToDesktop(content: String, fileName: String): Unit = {
```
Comment (suryabalasingh) [inline test/controllers/register/company/CheckYourAnswersControllerSpec.scala]: should be removed

**Pull request merged**